urwid/display_common.py

```diff
@@ -554,6 +556,13 @@ def __init__(self, fg: str, bg: str, colors: Literal[1, 16, 88, 256, 16777216] =
                 f'require more colors than have been specified ({colors:d}).'
             )
 
+    def __hash__(self) -> int:
+        """Safe hashable.
+
+         Instance is mutable -> use ID component.
+         """
+        return hash((self.__class__, self._value, id(self)))
```

Comment:
if `_value` is mutable should it be used for computing a `__hash__`? Won't this cause a problem if it's used as a key in a dict then modified?

Reply:
It's made like done by "unsafe hash" in python data classes. Making instance id specific (not recalculated) hash maybe also solution.
Hash is required for terminal example app.

Reply:
interesting, I didn't know about that. Is it worth mentioning this expected unsafe behavior or the reason for adding `__hash__` in the docstring?

Reply:
Immutable AttrSpec would be a good change